components/styled/table/src/molecules/table/table-core.tsx

```diff
@@ -0,0 +1,218 @@
+
+const columnHelper = createColumnHelper<DataRow>()
+
+export const DefaultTableOptions = {
+  showExpanderColumn: false,
+  pageSize: 10,
+  enablePagination: false,
+  serverSidePagination: false,
+  hidePaginationControls: false,
+  hidePerPageOptions: false,
+}
+
+export const DefaultPerPageOptions = [10, 20, 30, 40, 50]
+
+export const generateColumnsFromHeadersRecursively = (
+  headers: Header[],
+  showExpanderColumn: boolean
+): ColumnDef<DataRow, CellData | unknown>[] => {
+  const createColumns = (headers: Header[]): ColumnDef<DataRow, CellData | unknown>[] =>
+    headers?.map((header) =>
+      header.subheaders
+        ? columnHelper.group({
+            id: header.id ?? uuid(),
+            header: () => header.header,
+            columns: createColumns(header.subheaders ?? []),
+          })
+        : (columnHelper.accessor(header.accessor, {
+            id: header.accessor,
+            cell: ({ row, ...info }) => {
+              const stringValue = (info?.getValue() as string) ?? ''
+              const cellContent = header.cell
+                ? header.cell({ value: stringValue, columnId: info.column.id, row: row.original })
+                : stringValue
+
+              return row.getCanExpand() ? (
+                <span onClick={row.getToggleExpandedHandler()} style={{ cursor: 'pointer' }}>
+                  {cellContent}
+                </span>
+              ) : (
+                cellContent
+              )
+            },
+            header: () => header.header,
+          }) as ColumnDef<DataRow, CellData | unknown>)
+    ) ?? []
+
+  const tableColumns = createColumns(headers ?? [])
+
+  return showExpanderColumn && tableColumns.length > 0 ? [getExpanderColumn(), ...tableColumns] : tableColumns
+}
+
+export const generateRowsFromCellRows = (cellRows: CellRow[]): DataRow[] =>
+  cellRows.map((cellRow) => {
+    const mappedCell: DataRow = { subRows: [] }
+    const mappedCellRender: Record<string, FC<ICellProps>> = {}
+
+    cellRow.cells.forEach((cell) => {
+      mappedCell[cell.key] = cell.value
+      if (cell.render) {
+        mappedCellRender[cell.key] = cell.render
+      }
+    })
+
+    return {
+      ...mappedCell,
+      ...{
+        rowId: cellRow.id ?? '',
+        render: cellRow.render ? cellRow.render : (props: ICellProps) => <>{props.value}</>,
+        renderCells: mappedCellRender,
+      },
+      subRows: generateRowsFromCellRows(cellRow.subRows ?? []),
+    }
+  })
+
+export const getExpanderColumn = (): ColumnDef<DataRow, CellData | unknown> =>
+  columnHelper.accessor('expander', {
+    header: ({ table }) => (
+      <span
+        title="Toggle All Rows Expanded"
+        onClick={table.getToggleAllRowsExpandedHandler()}
+        style={{ cursor: 'pointer' }}
+      >
+        {table.getIsAllRowsExpanded() ? '▲' : '►'}
+      </span>
+    ),
+    cell: ({ row }) =>
+      row.getCanExpand() ? (
+        <span
+          {...{
```

Comment:
I'm not sure this spread operator is necessary - also as per my comments on the icons, I think they should be buttons if they're clickable